src/dashboard/src/main/management/commands/rebuild_transfer_backlog.py

```diff
@@ -0,0 +1,209 @@
+        else:
+            self.success('Connected to Elasticsearch node {} (v{}).'.format(
+                es_info['name'], es_info['version']['number']))
+
+        self.delete_index(es_client)
+        self.create_index(es_client)
+        self.populate_index(es_client, transfer_backlog_dir)
+        self.success('Indexing complete!')
+
+    def confirm(self, no_prompt):
+        """Ask user to confirm the operation."""
+        if no_prompt:
+            return True
+        self.stdout.write('WARNING: This script will delete your current '
+                          'Elasticsearch transfer data, rebuilding it using '
+                          'files.')
+        return boolean_input('Are you sure you want to continue?')
+
+    def prepdir(self, path):
+        """Append ''originals'' to the path if it's missing."""
+        tail = os.path.basename(os.path.normpath(path))
+        suffix = 'originals'
+        if tail != suffix:
+            return os.path.join(path, suffix)
+
+    def delete_index(self, es_client):
+        """Delete search index."""
+        self.stdout.write('Deleting all transfers in the "transfers" index...')
+        time.sleep(3)  # Time for the user to panic and kill the process.
+        es_client.indices.delete('transfers', ignore=404)
+
+    def create_index(self, es_client):
+        """Create search index."""
+        self.stdout.write('Creating index...')
+        elasticSearchFunctions.create_transfers_index(es_client)
+
+    def populate_index(self, es_client, transfer_backlog_dir):
+        """Populate search index."""
+        for directory in os.listdir(transfer_backlog_dir):
+            if directory == '.gitignore':
+                continue
+
+            transfer_uuid = directory[-36:]
+            transfer_dir = os.path.basename(directory)
+            self.stdout.write('Indexing {} ({})'.format(
+                transfer_uuid, transfer_dir))
+
+            self.process_transfer(es_client, os.path.join(transfer_backlog_dir, directory), transfer_uuid)
+            self.ss_reindex(transfer_uuid)
+
+    def process_transfer(self, es_client, transfer_path, transfer_uuid):
+        transfer_dir = os.path.basename(transfer_path)
+        tree = ElementTree.parse(os.path.join(
+            transfer_path, 'metadata/submissionDocumentation/METS.xml'))
+        root = tree.getroot()
+
+        # Attempt to insert row in MySQL Transfers table if it doesn't already
+        # exist.
+        try:
+            Transfer.objects.create(
+                uuid=transfer_uuid,
+                currentlocation=os.path.join(
+                    transfer_path.replace(django_settings.SHARED_DIRECTORY,
+                                          '%sharedPath%', 1),
+                    transfer_dir))
+        except django.db.utils.IntegrityError:
+            pass
+
+        # Extract list of files from METS data so as to populate the MySQL
+        # Files table.
+        mets_bns = '{http://www.loc.gov/METS/}'
```

Comment:
Maybe this should be taken from archivematicaCommon/lib/namespaces.py so it is centralized. Same with xlink.